server/storage/backend/backend.go

```diff
@@ -445,6 +455,27 @@ func (b *backend) Defrag() error {
 	return b.defrag()
 }
 
+func (b *backend) SubscribeDefragNotifier(notifier DefragNotifier) {
+	if notifier == nil {
+		return
+	}
```

Comment:
Why this check? It doesn't protect against nil pointer exceptions.

It will work for 
```
b.SubscribeDefragNotifier(nil)
```
But not for 
```
var notifier *healthNotifier
b.SubscribeDefragNotifier(notifier)
```

Because `nil != (*healthNotifier)(nil)`

Reply:
this is to protect the line
```
mc := adapter.MaintenanceServerToMaintenanceClient(v3rpc.NewMaintenanceServer(s, nil))
```

Also, why would not it work for 
```
var notifier *healthNotifier
b.SubscribeDefragNotifier(notifier)
```

I tried 
```
type Person struct {
	Name string
	Age  int
}

var persons []*Person

func subscribePerson(p *Person) {
	if p == nil {
		fmt.Printf("p is nil\n")
	} else {
		fmt.Printf("p is not nil\n")
		persons = append(persons, p)
	}
}

func main() {
	var p1 *Person
	subscribePerson(p1)
	fmt.Printf("len(persons) = %d\n", len(persons))
}
```
It works with output `len(persons) = 0`